### DISCUSSION TITLE
[BUG] Custom Transformation issue with Atlasmap

### DISCUSSION THREAD

**Discussion by Shrishti1234** [atlasmap/atlasmap#3456] [Q&A]
**Describe the bug**
I tried to use custom transformation of altasmap with 2.2.3 version. Custom transformation jar imported fine but mapping definition is null while doing XML to JSON transformation. I also tried to use 2.3.1 version but with this version Custom jar is imported but transformation is not present in drop down list.

**Attach .adm file and source document examples**
I have attached zip file which have following files.

1.Atlasmap demo project - (XmlToJsonTransformation) class which is used for conversion
2.Custom project (for creating custom transformation)
3.altas-map.adm ( mapping definition file for transformation)
4.orderxml.xsd ( sample request for conversion XML to JSON
 
**To Reproduce**
1. Go to atlasmap gui 
2. Click on Inport instance or schema file and then import import attached order.json file or any other intstance in source panel and other instance in target panel
3. After that import jar archive for custom transformation. ( I have attached my custom transformation project).
4. Apply some custom transformation on target element.
5. Export .adm file and use this mapping file for transformation.
6. Check mapping object in session object it is null. while doing transformation

**Expected behavior**
Mapping objet should not be null. I shoud not get mapping definition null. Transformation should work fine

**Environment (please complete the following information):**
- OS: [e.g. Windows]
 - Browser [Microsoft Edge]
 - Version [ 88.0.705.81]

[Custom Transformation.zip](https://github.com/atlasmap/atlasmap/files/7396162/Custom.Transformation.zip)
)

**Comment by igarashitm** [comment]
Please follow the issue template and describe what is the problem, for example steps to reproduce, what is the expected behavior vs. what is the actual behavior you see, etc.

**Comment by igarashitm** [comment]
Also please put a title describing about the bug in one sentence.

**Comment by Shrishti1234** [comment]
> Please follow the issue template and describe what is the problem, for example steps to reproduce, what is the expected behavior vs. what is the actual behavior you see, etc.


done @igarashitm

**Comment by igarashitm** [comment]
What do you mean by `Mapping objet` and `mapping definition`? are they different things or referring same one? what is the instance source document you send at runtime? is that orderxml.xsd an instance, but not a schema you load into UI at design time? are you mapping XSD file? How did you test `Check mapping object in session object it is null. while doing transformation
`? what do you mean by `session object`? is it same with `Mapping objet` and/or `mapping definition`?

In AtlasMap word, mapping definition is a JSON file contained in .adm file, which defines how to process mapping. Do you mean you extracted the JSON file from .adm file and looked at that? If so, what do you mean by `get mapping definition null`? there was no mapping definition file in the .adm? or there was `null` in the file?

**Comment by Shrishti1234** [comment]
> What do you mean by `Mapping objet` and `mapping definition`? are they different things or referring same one? what is the instance source document you send at runtime? is that orderxml.xsd an instance, but not a schema you load into UI at design time? are you mapping XSD file? How did you test `Check mapping object in session object it is null. while doing transformation `? what do you mean by `session object`? is it same with `Mapping objet` and/or `mapping definition`?
> 
> In AtlasMap word, mapping definition is a JSON file contained in .adm file, which defines how to process mapping. Do you mean you extracted the JSON file from .adm file and looked at that? If so, what do you mean by `get mapping definition null`? there was no mapping definition file in the .adm? or there was `null` in the file?

Hi @igarashitm ,
Mapping Object and Mapping definition both are same thing. In GUI I have load XML instance i.e. orderxml but the issue is not with source document. I am attaching one zip file of screenshots.
1. **Customtransformationmapping** is screenshot of mapping file opened in GUI which is exported and use this .adm file in projet.
2. **MappingObjectNull** is screenshot of mapping object which is highlighted is showing null and because of that I get an error mapping definition can't be null.
3. **CodeForTransformation** is screenshot of code which is used for conversion. **It is working fine without custom transformation
[Custom transformation images.zip](https://github.com/atlasmap/atlasmap/files/7465124/Custom.transformation.images.zip)
**

**Comment by igarashitm** [comment]
Moved to a discussion as this doesn't describe one specific issue.
BTW Edge is not a supported browser. Can you try with Chrome or Firefox? Also try with latest 2.3.2?

**Comment by danlz** [comment]
This is the reason:
https://github.com/atlasmap/atlasmap/blob/main/lib/core/src/main/java/io/atlasmap/core/DefaultAtlasContext.java#L163
IgnoreLibrary is always true, so the custom transformations are never loaded.

**Reply by arockiaraj1994** [reply]
Is this sorted out? I am facing the same issue. Unable to make the custom transformation
